images/egress/dns-proxy/egress-dns-proxy.sh

```diff
@@ -58,6 +58,9 @@ defaults
     timeout server          1m
     timeout http-keep-alive 100s
     timeout check           10s
+
+    # Never fail on address resolution
+    default-server init-addr last,libc,none
```

Comment:
`none` essentially constitutes failure since if it can't resolve its address at startup, it's not going to resolve it later either, is it?

Is there some bug report that this PR corresponds to?

Reply:
Yes, `none` is meant for 'do not bring haproxy down'  if the resolution fails during startup.
We support multiple destinations and the original intention was one destination should not bring down the entire egress router.
As @danwinship pointed, if there is a genuine error in dns name or if the name never resolve, there will an Alert/Warning in the log but user may never notice.
Another way to solve resolution failure during startup is to use restart on failure policy for the egress pod and if it is a transient issue subsequent restarts should fix it and in case of genuine issue it will fail repeated and will have CrashLoopBackOff status. This seems a reasonable solution, so I will remove the `none` option.

Reply:
> We support multiple destinations and the original intention was one destination should not bring down the entire egress router.

Ah, I don't think this would fix that anyway. According to the docs, `default-server init-addr ...` is only about the pod resolving *its own* address, not about resolving the addresses of other hosts mentioned in the config.